packages/react-components/src/components/chart/addTrendCursor.ts

```diff
@@ -0,0 +1,179 @@
+import { v4 as uuid } from 'uuid';
+import {
+  DEFAULT_MARGIN,
+  trendCursorCloseButtonXOffset,
+  trendCursorCloseButtonYOffset,
+  trendCursorHeaderBackgroundColor,
+  trendCursorHeaderColors,
+  trendCursorHeaderTextColor,
+  trendCursorHeaderWidth,
+  trendCursorLineColor,
+  trendCursorLineWidth,
+  trendCursorMarkerRadius,
+  trendCursorZIndex,
+} from './eChartsConstants';
+import { EChartsType, ElementEvent, LineSeriesOption, SeriesOption } from 'echarts';
+import { ChartEventType, InternalGraphicComponentGroupOption, SizeConfig } from './types';
+import close from './close.svg';
+import { Dispatch, SetStateAction } from 'react';
+import { GraphicComponentTextOption } from 'echarts/types/src/component/graphic/GraphicModel';
+import { Viewport } from '@iot-app-kit/core';
+import {
+  calculateTimeStamp,
+  calculateTrendCursorsSeriesMakers,
+  getTrendCursorHeaderTimestampText,
+  setXWithBounds,
+} from './utils/getInfo';
+
+// this returns a Graphic element of Echarts (https://echarts.apache.org/en/option.html#graphic)
+// A Trend cursor is a custom Graphic group element,
+// which has a line and other elements which gets rendered on the screen.
+// for now, we are storing the timestamp of the trend cursor in graphic element
+// which will eventually be moved to state(redux)
+const addNewTrendCursor = (
+  e: ElementEvent,
+  size: SizeConfig,
+  count: number,
+  graphic: InternalGraphicComponentGroupOption[],
+  setGraphic: Dispatch<SetStateAction<InternalGraphicComponentGroupOption[]>>,
+  series: SeriesOption[],
+  yMax: number,
+  yMin: number,
+  viewport?: Viewport,
+  chart?: EChartsType
+) => {
+  const uId = uuid();
+  // TODO: test this once echarts live mode is supported
+  const timestampInMs = calculateTimeStamp(e.offsetX, size.width, viewport);
+  const boundedX = setXWithBounds(size, e.offsetX ?? 0);
+  // TODO: test this once echarts live mode is supported
+  const yAxisMarkers = calculateTrendCursorsSeriesMakers(series, yMax, yMin, timestampInMs, size.height);
+  const newTC = {
```

Comment:
could we break different sections of this out into separate functions?

its a 100+ line object and it feels easy to get lost in whats going on

Reply:
sure, was trying to avoid jumping into multiple files.